File: frontend-api/src/main/java/no/nav/melosys/tjenester/gui/dto/MedlemskapsperiodeDto.java
package no.nav.melosys.tjenester.gui.dto;

import java.time.LocalDate;

import no.nav.melosys.domain.InnvilgelsesResultat;
import no.nav.melosys.domain.Medlemskapsperiode;
import no.nav.melosys.domain.kodeverk.Folketrygdloven_kap2_bestemmelser;
import no.nav.melosys.domain.kodeverk.Medlemskapstyper;
import no.nav.melosys.domain.kodeverk.Trygdedekninger;

public record MedlemskapsperiodeDto (long id,
                                     String arbeidsland,
                                     LocalDate fomDato,
                                     LocalDate tomDato,
                                     Folketrygdloven_kap2_bestemmelser bestemmelse,
                                     InnvilgelsesResultat innvilgelsesResultat,
                                     Trygdedekninger trygdedekning,
                                     Medlemskapstyper medlemskapstype) {

    public long getId() {
        return id;
    }

    public String getArbeidsland() {
        return arbeidsland;
    }

    public LocalDate getFomDato() {
        return fomDato;
    }

    public LocalDate getTomDato() {
        return tomDato;
    }

    public Folketrygdloven_kap2_bestemmelser getBestemmelse() {
        return bestemmelse;
    }

    public InnvilgelsesResultat getInnvilgelsesResultat() {
        return innvilgelsesResultat;
    }

    public Trygdedekninger getTrygdedekning() {
        return trygdedekning;
    }

    public Medlemskapstyper getMedlemskapstype() {
        return medlemskapstype;
    }

    public static MedlemskapsperiodeDto av(Medlemskapsperiode medlemskapsperiode) {
        return new MedlemskapsperiodeDto(
            medlemskapsperiode.getId(),
            medlemskapsperiode.getArbeidsland(),
            medlemskapsperiode.getFom(),
            medlemskapsperiode.getTom(),
            medlemskapsperiode.getBestemmelse(),
            medlemskapsperiode.getInnvilgelsesresultat(),
            medlemskapsperiode.getDekning(),
            medlemskapsperiode.getMedlemskapstype()
        );
    }
}

File: integrasjon/src/main/kotlin/no/nav/melosys/integrasjon/trygdeavgift/dto/MedlemskapsperiodeDto.kt
package no.nav.melosys.integrasjon.trygdeavgift.dto

import no.nav.melosys.domain.Medlemskapsperiode
import no.nav.melosys.domain.kodeverk.Avgiftsdekning
import no.nav.melosys.domain.kodeverk.Medlemskapstyper
import java.util.*


data class MedlemskapsperiodeDto(
    val id: UUID,
    val periode: DatoPeriodeDto,
    val avgiftsdekninger: Set<Avgiftsdekning>,
    val medlemskapstype: Medlemskapstyper,
) {
    companion object {

        fun Medlemskapsperiode.idToUUID(): UUID {
            return idToUUid(this.id)
        }

        private fun idToUUid(id: Long): UUID {
            return UUID.nameUUIDFromBytes(id.toString().toByteArray())
        }
    }
}
